synfig-studio/src/synfigapp/actions/waypointsetsmart.cpp

```diff
@@ -111,10 +112,12 @@ bool
 Action::WaypointSetSmart::is_candidate(const ParamList &x)
 {
 	return (candidate_check(get_param_vocab(),x) &&
-			// We need an animated valuenode.
-			ValueNode_Animated::Handle::cast_dynamic(x.find("value_node")->second.get_value_node()) &&
 			// We need either a waypoint or a time.
-			(x.count("waypoint") || x.count("time")));
+			(x.count("waypoint") || x.count("time")) &&
+			// Parameter has to be animatable
+			!(x.find("value_desc")->second.get_value_desc()).get_static() &&
```

Comment:
You can replace `x.find("value_desc")->second` with `x.at("value_desc")`

Reply:
x (ParamList) is a multimap, so we can't really use `at()` for element access here.

Reply:
Oh, I forgot this is a multimap :+1: